diff --git a/Utilities/cmcurl/setup.h b/Utilities/cmcurl/setup.h
index fff6974711..d2fafc6c01 100644
--- a/Utilities/cmcurl/setup.h
+++ b/Utilities/cmcurl/setup.h
@@ -63,7 +63,6 @@
 #endif
 
 #if !defined(__cplusplus) && !defined(__BEOS__)
-typedef unsigned char bool;
 #define typedef_bool
 #endif
 
diff --git a/Utilities/cmxmlrpc/bool.h b/Utilities/cmxmlrpc/bool.h
index 8bf54f34f7..b2a365182f 100644
--- a/Utilities/cmxmlrpc/bool.h
+++ b/Utilities/cmxmlrpc/bool.h
@@ -11,7 +11,6 @@
 #ifndef __cplusplus
 #ifndef HAVE_BOOL
 #define HAVE_BOOL
-typedef int bool;
 #endif
 #endif
 
