diff --git a/Source/CTest/cmCTestBuildHandler.cxx b/Source/CTest/cmCTestBuildHandler.cxx
index 9678ac45c0..baf376760c 100644
--- a/Source/CTest/cmCTestBuildHandler.cxx
+++ b/Source/CTest/cmCTestBuildHandler.cxx
@@ -568,7 +568,7 @@ class cmCTestBuildHandler::FragmentCompare
 public:
   FragmentCompare(cmFileTimeComparison* ftc): FTC(ftc) {}
   FragmentCompare(): FTC(0) {}
-  bool operator()(std::string const& l, std::string const& r)
+  bool operator()(std::string const& l, std::string const& r) const
     {
     // Order files by modification time.  Use lexicographic order
     // among files with the same time.
