diff --git a/Source/CTest/cmCTestBuildHandler.cxx b/Source/CTest/cmCTestBuildHandler.cxx
index 0095bbc..6670e7e 100644
--- a/Source/CTest/cmCTestBuildHandler.cxx
+++ b/Source/CTest/cmCTestBuildHandler.cxx
@@ -529,7 +529,7 @@ class cmCTestBuildHandler::FragmentCompare
 {
 public:
   FragmentCompare(cmFileTimeComparison* ftc): FTC(ftc) {}
-  bool operator()(std::string const& l, std::string const& r)
+  bool operator()(std::string const& l, std::string const& r) const
     {
     // Order files by modification time.  Use lexicographic order
     // among files with the same time.
